motor/providers/k8s/manifest_parser.go

```diff
@@ -437,5 +468,31 @@ func load(manifest []byte) ([]k8sRuntime.Object, error) {
 		res = append(res, objects...)
 	}
 
+	resList, err := resources.CachedServerResources()
+	if err != nil {
+		return nil, err
+	}
+
+	resTypes, err := resources.ResourceIndex(resList)
+	if err != nil {
+		return nil, err
+	}
+
+	// Every unstructured object here is an object that we couldn't match to an actual type.
+	// Such objects we treat as custom resources and should end up in the k8s.customresources list.
+	// To do that we need to have a CRD for every kind that couldn't be matched to a type. Here we create
+	// the related CRD for every type that needs it.
+	for _, o := range res {
+		if unstr, ok := o.(*unstructured.Unstructured); ok {
+			gvk := unstr.GetObjectKind().GroupVersionKind()
+			if _, err := resTypes.Lookup(gvk.Kind); err != nil {
+				crdName := strings.ToLower(fmt.Sprintf("%s.%s", gvk.Kind, gvk.Group))
+				res = append(
```

Comment:
I'm not sure here:

You are iterating over `res` and append to `res`. Don't we have the object now twice?

Reply:
No because the code only adds entries to the list if it sees an `Unstructured` object. Since we are not adding `Unstructured` objects, the code won't run twice. However, this made me see that we can add the same CRD twice if, for example, we scan a manifest that contains 2 times the same resource. I fixed that now